### DISCUSSION TITLE
Docker Image

### DISCUSSION THREAD

**Discussion by leoafarias** [leoafarias/fvm#633] [General]
There are a few open issues about Docker images in FVM, and I wanted to consolidate them in a discussion. But it's important to align a few things.

- The current image defaults to the latest on `dart pub global` which is neither ideal nor version specific. 
- I have worked on a [install script](https://github.com/leoafarias/fvm/blob/main/scripts/install.sh) that I think should be the best solution from now on in many cases, and I want to switch to that.

It would be good to discuss the requirement for this image and if anyone can help get a PR for this. cc @deandreamatias @AngeloAvv @SebastianKlaiber

**Comment by AngeloAvv** [comment]
Hey @leoafarias , from my point of view, we can easily satisfy both of your requirements.

> * The current image defaults to the latest on `dart pub global` which is neither ideal nor version specific.

It's okay to have a :latest version, but it is also important to have different images pointing to a specific version, this could be easily achieved by tagging a specific image version every time you release a new fvm version. 

> * I have worked on a [install script](https://github.com/leoafarias/fvm/blob/main/scripts/install.sh) that I think should be the best solution from now on in many cases, and I want to switch to that.

We can create a Dockerfile that prepares the base environment and then runs the script you created. The only thing that comes to my mind that should be implemented (if it's not already there) is the non-interactivity of the script itself. Apart from that, we can ship a docker image with a specific fvm version compiled using the new script

**Reply by leoafarias** [reply]
That makes sense. However, some Dockerfiles for Flutter include other utilities and dependencies like Android tools and others, while the current Dockerfile is very simple. Does this work? Not sure what are the current needs are.

**Reply by SebastianKlaiber** [reply]
How about using a similiar approch to cirlce ci? A Base image and some more specified with android ...

https://github.com/CircleCI-Public/cimg-base
https://github.com/CircleCI-Public/cimg-android

**Reply by AngeloAvv** [reply]
@leoafarias @SebastianKlaiber I created an image for specific Android builds a few months ago which includes fvm and other tools

https://github.com/MyLittleSuite/mobile_ci

**Comment by deandreamatias** [comment]
I think that some requirements would be:
1. A image published on DockerHub
2. This image, should be tagged on DockerHub. One tag for each FVM version. 
3. To accomplish the point 2, can automatize the process for Github Actions.

**Reply by leoafarias** [reply]
@deandreamatias can you do a PR for an initial image using the install.sh? I think the versioned tags can be fairly easy after that

**Reply by deandreamatias** [reply]
I don't sure how do that, but would be investigate this weekend

**Reply by deandreamatias** [reply]
I made my first steps: https://github.com/deandreamatias/fvm/commit/569a3b74685fe8cd3e6eecc95bca6d2c385cce11

**Reply by leoafarias** [reply]
@deandreamatias, Thanks for that; open a PR, and I can leave some comments on it.

**Reply by deandreamatias** [reply]
I will try setup the pipeline for automatic release too

**Comment by SebastianKlaiber** [comment]
I also think a Docker image that is published with each fvm version at Dockerhub makes sense then customized Docker images like mine that requires a SQLite instance could use the fvm as base image. 

@leoafarias  What are the requirements for the script to run? Does Dart need to be installed? ...

**Reply by SebastianKlaiber** [reply]
@leoafarias unfortunately, the link in the install.md leads to a 404 page. `curl -fsSL https://fvm.app/install.sh | bash`

**Reply by leoafarias** [reply]
@SebastianKlaiber I have not deployed the script yet, but you can get from github like this

`curl -fsSL https://raw.githubusercontent.com/leoafarias/fvm/main/scripts/install.sh | bash`

**Reply by leoafarias** [reply]
@SebastianKlaiber I have just updated the script, also I created an uninstall script

**Comment by deandreamatias** [comment]
@leoafarias currently the pipelines generate images on each version right? I see a new docker image version here https://hub.docker.com/r/leoafarias/fvm/tags
Maybe can close this discussion
